### PR TITLE
Lets xenos drag and drop themselves into a vent crawl

### PR DESCRIPTION
**Pull request by GrrrKitten**
# About the pull request

Pt1 of making vents cooler to use

you click ur xeno -> u drag ur mouse while holding m1 -> u drop it on da vent -> badda bing bodda boom ur starting to crawl into da vent

# Explain why it's good for the game

QOL, it feels like this is just how it should be, not sure why its not like this already

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

image

image

</details>

---

Like all the cool stuff I make?
[ko-fi](https://ko-fi.com/J3J6102QI0)

# Changelog

:cl: Grrr Kitten
qol: xenos can now drag and drop themselves to vent crawl
/:cl:

### CODE REVIEW

**Comment by RainbowStalin** [comment]
Finaly a way to get into vents without using the verbs.

**Comment by TheManWithNoHands** [comment]
> Finaly a way to get into vents without using the verbs.

Verb? Isnt it just alt + M1.

**Comment by GrrrKitten** [comment]
> > Finaly a way to get into vents without using the verbs.
> 
> 
> 
> Verb? Isnt it just alt + M1.

I didn't know this, why must there be so much esoteric knowledge behind dumb binds. 

Anyways I'm keeping this up as it doesn't hurt to have more options

**Comment by vero5123** [comment]
~money~

File: code/game/objects/structures/pipes/vents/vents.dm [hunk 1]
```diff
@@ -70,6 +70,12 @@
 /obj/structure/pipes/vents/hide()
 	update_underlays()
 
+/obj/structure/pipes/vents/MouseDrop_T(obj/structure/pipes/vents/dropvent, mob/living/carbon/xenomorph/user)
+	if(!isxeno(user) || !user.check_state() || !user.can_ventcrawl())
+		return
+	if(dropvent in range(1))
```
Comment (harryob) [inline code/game/objects/structures/pipes/vents/vents.dm:76]:
```suggestion
	if(get_dist(src, user) <= 1)
```

File: code/game/objects/structures/pipes/vents/vents.dm [hunk 2]
```diff
@@ -70,6 +70,12 @@
 /obj/structure/pipes/vents/hide()
 	update_underlays()
 
+/obj/structure/pipes/vents/MouseDrop_T(obj/structure/pipes/vents/dropvent, mob/living/carbon/xenomorph/user)
```
Comment (harryob) [inline code/game/objects/structures/pipes/vents/vents.dm:73]: isn't dropvent here the atom being dropped (aka the user)?

```suggestion
/obj/structure/pipes/vents/MouseDrop_T(atom/dropping, mob/living/carbon/xenomorph/user)
```

File: code/game/objects/structures/pipes/vents/vents.dm [hunk 3]
```diff
@@ -70,6 +70,12 @@
...
+	if(!isxeno(user) || !user.check_state() || !user.can_ventcrawl())
```
Comment (harryob) [inline code/game/objects/structures/pipes/vents/vents.dm:74]:
```suggestion
	if(!isxeno(user) || !user.check_state() || !user.can_ventcrawl() || dropping != user)
```

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by Drulikar** [comment]
Feel free to ask for reopen when above can be addressed.

**Pull request closed without merging**